File: Easy/1018 - BinaryPrefixDivisibleBy5/Solution.cs
// Question 1018. Binary Prefix Divisible By 5 (https://leetcode-cn.com/problems/binary-prefix-divisible-by-5/)

/// <summary>
/// 执行用时：316 ms, 在所有 C# 提交中击败了 31.58% 的用户
/// 内存消耗：34.1 MB, 在所有 C# 提交中击败了 68.42% 的用户
/// 
/// Acknowledgement: This solution borrows idea from the [official solution](https://leetcode-cn.com/problems/binary-prefix-divisible-by-5/solution/ke-bei-5-zheng-chu-de-er-jin-zhi-qian-zh-asih/)
/// </summary>
public class Solution 
{
    public IList<bool> PrefixesDivBy5(int[] A) 
    {
        int N = A.Count();
        IList<bool> divisibleArray = new List<bool>();
        
        List<int> subArray = new List<int>();
        int value = 0;
        for (var i = 0; i < N; i++)
        {
            value = ((value << 1) + A[i]) % 5;
            divisibleArray.Add(value == 0);
        }

        return divisibleArray;
    }
}

File: Easy/1018 - BinaryPrefixDivisibleBy5/solution.ts
// Question 1018. Binary Prefix Divisible By 5 (https://leetcode-cn.com/problems/binary-prefix-divisible-by-5/)

/**
 * 执行用时：96 ms, 在所有 TypeScript 提交中击败了 100.00% 的用户
 * 内存消耗：42.7 MB, 在所有 TypeScript 提交中击败了 100.00% 的用户
 *
 * @param {number[]} A
 * @returns {boolean[]}
 */
function prefixesDivBy5(A: number[]): boolean[] {
  const results: boolean[] = [];
  const N: number = A.length;

  let value: number = 0;
  for (let index = 0; index < N; index++) {
    value = ((value << 1) + A[index]) % 5;
    results.push(!value);
  }

  return results;
}
